use-intersection-observer.js

```diff
@@ -0,0 +1,52 @@
+import { useState, useEffect } from "react";
+
+const INTERSECTION_OBSERVER_CONFIG = {
+  threshold: 0.0,
+  triggerOnce: true,
+  rootMargin: "0px"
+};
+
+/**
+ * React hook that return is the object is intersected
+ */
+function useIntersectionObserver(ref, config = {}, rootRef = null) {
+  const [isIntersecting, setIntersecting] = useState(false);
+
+  const options = {
+    root: rootRef && rootRef.current,
+    ...INTERSECTION_OBSERVER_CONFIG,
+    ...config
+  };
+
+  useEffect(() => {
+    const observer = new IntersectionObserver(function(entries) {
+      if (options.triggerOnce) {
+        if (entries[0].isIntersecting) {
+          setIntersecting(true);
+          observer.unobserve(ref.current);
+        }
+      } else {
+        setIntersecting(entries[0].isIntersecting);
+      }
+    }, options);
+
+    if (!ref) {
+      console.warn(
+        `No target specified - use useRef() hook to pass the target value`
+      );
+      return;
+    }
+
+    const { current } = ref;
+
+    observer.observe(current);
+
+    return () => {
+      if (current) observer.unobserve(current);
+    };
+  }, [options, ref, options.triggerOnce]);
```

Comment:
@alemesa careful with the options, since in line 15 you are creating a new options object with each render, so it would recreate the observers and so on due to it being a new object each time. You might want to try `[config, ref, options.triggerOnce]`